### PR TITLE
feature/MS-988/ audio response in lti outcome UI

### PR DESCRIPTION
**Pull request by anshul119**
Ticket: https://oat-sa.atlassian.net/browse/MS-988

- Added the file response in ltiOutcomeUi

### How to test?

- Create an audio PCI item
- Create a test, publish it  and take the test
- Create an LTI provider
- Go manage roles from the top-right menu > LTI Roles > Instructor > select Test Taker role and save
- Go to http://ltiapps.net/test/tc.php and add the launch URL(`https://YOUR_DOMAIN/ltiOutcomeUi/ItemResultPreviewer/launch?resultId=YOUR_RESULT_URI&itemRef=YOU_ITEM_REF`), key and secret of the LTI provider, add the `instructor` role and launch it
- Audio PCI should work as expected in review mode with LTI

### CODE REVIEW

File: views/js/controller/ItemResultPreviewer/itemResultPreviewer.js [hunk 1]
```diff
@@ -12,8 +21,68 @@ define(['taoItems/previewer/factory'], function(previewerFactory) {
...
+    }
+
+    /**
+     * Makes sure the response contains the file data if it is a file record
+     * @param {Object} response
+     * @param {String} deliveryUri
+     * @returns {Promise}
+     */
+    function refineFileResponse(response, deliveryUri) {
+        const { file } = (response && response.base) || {};
```
Comment (dresha-dev) [inline views/js/controller/ItemResultPreviewer/itemResultPreviewer.js]:
```suggestion
        const file = response && response.base && response.base.file;
```

File: views/js/controller/ItemResultPreviewer/itemResultPreviewer.js [hunk 2]
```diff
@@ -12,8 +21,68 @@ define(['taoItems/previewer/factory'], function(previewerFactory) {
...
+    function refineItemState(state, deliveryUri) {
+        if (!state) {
+            return Promise.resolve(state);
+        }
+
+        const filePromises = Object.keys(state).map(identifier => {
+            const { response } = state[identifier];
+            return refineFileResponse(response, deliveryUri);
+        });
+        return Promise.all(filePromises).then(() => state);
```
Comment (dresha-dev) [inline views/js/controller/ItemResultPreviewer/itemResultPreviewer.js:81]: Should we handle errors?

File: views/js/loader/ltiOutcomeUi.min.js [hunk 3]
```diff
@@ -1,2 +1,2 @@
...
...t.keys(state).map(function(identifier){var response=state[identifier].response;return refineFileResponse(response,deliveryUri)});return Promise.all(filePromises).then(function(){return state})}var logger=loggerFactory("ltiOutcomeUi/ItemResultPreviewer"),dataUrl=urlHelper.route("getVariableFile","ItemResultPreviewer","ltiOutcomeUi"),plugins=[{module:"ltiOutcomeUi/previewer/plugins/content/scoringHighlighter",bundle:"ltiOutcomeUi/loader/ltiOutcomeUi.min",category:"content"}];return{start:function start(config){var uri=config.uri,resultIdentifier=config.resultIdentifier,itemDefinition=config.itemDefinition,deliveryUri=config.deliveryIdentifier,uri={uri:uri,resultId:resultIdentifier,itemDefinition:itemDefinition,deliveryUri:deliveryUri},type=config.type;refineItemState(config.state,deliveryUri).then(function(state){previewerFactory(type,uri,state,{view:"scorer",fullPage:!0,hideActionBars:!0,plugins:plugins})})}}}),define("ltiOutcomeUi/controller/routes",{ItemResultPreviewer:{actions:{preview:"controller/ItemResultPreviewer/itemResultPreviewer"}}}),define("ltiOutcomeUi/loader/ltiOutcomeUi.bundle",function(){}),window.bundles=(window.bundles||[]).concat(["taoItems/loader/taoItems.min"]);
```
Comment (dresha-dev) [inline views/js/loader/ltiOutcomeUi.min.js:1]: Are the bundles must be creating during the release?

Comment (anshul119) [reply views/js/loader/ltiOutcomeUi.min.js:1]: there is no harm to do it before hand as I know

**Review by jsconan — APPROVED** [review]
Note: The Test Taker role needs to be added to the LTI Instructor role for the ltiOutcomeUi to access the PCI
image


**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by dresha-dev — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request's target is not `master`
- [x] Pull request title and description are meaningful
- [x] Commits are following conventional commits

**Pull request merged**